feat: #6 create admin request management page

nevence opened the pull request:
created HR (admin) and Employee request management pages, implemented filtering by leave type, status, search by name, and page size 
implemented UI and logic for requests approval and rejection, and adding (editing) comments (HR and Employee) / changing leave type (HR)
![Screenshot 2024-06-19 020929](https://github.com/nevence/VacaYAY/assets/155846118/66e7f816-38b5-45ea-9ccb-ac500baaa8c1)
![Screenshot 2024-06-19 020942](https://github.com/nevence/VacaYAY/assets/155846118/c34bb68b-f07a-4405-ba35-17201a36c964)
![Screenshot 2024-06-19 020957](https://github.com/nevence/VacaYAY/assets/155846118/eedd9757-24d0-4a66-8f64-6d88b141d95b)
![Screenshot 2024-06-19 021044](https://github.com/nevence/VacaYAY/assets/155846118/b7f7003d-55c4-446b-80a3-8908a36610b2)
![Screenshot 2024-06-19 021114](https://github.com/nevence/VacaYAY/assets/155846118/dc093958-54fb-4e87-8655-8648b01f99fb)
![Screenshot 2024-06-16 150247](https://github.com/nevence/VacaYAY/assets/155846118/e201464b-655b-4471-9eb0-f874bb81ab59)
![Screenshot 2024-06-16 150304](https://github.com/nevence/VacaYAY/assets/155846118/1fcd392b-8d0e-471f-8e44-e7c0b63dfa9c)
![Screenshot 2024-06-16 150326](https://github.com/nevence/VacaYAY/assets/155846118/f3450602-7ba4-4772-8bde-1935319727b3)
![Screenshot 2024-06-16 150407](https://github.com/nevence/VacaYAY/assets/155846118/badb0cc7-fb58-4a8e-91ec-e8b878d28855)
![Screenshot 2024-06-16 150448](https://github.com/nevence/VacaYAY/assets/155846118/2508e7c1-54b3-4539-86b4-da3da75acd85)
![Screenshot 2024-06-16 150506](https://github.com/nevence/VacaYAY/assets/155846118/91806db1-2c9c-4167-986e-24b67f8d3f9f)
![Screenshot 2024-06-16 150532](https://github.com/nevence/VacaYAY/assets/155846118/d04f33da-1db6-4498-88e2-c0a73a75b5e2)
![Screenshot 2024-06-16 150546](https://github.com/nevence/VacaYAY/assets/155846118/ca3eacc7-e53d-4730-a514-cbc2703569e3)
![Screenshot 2024-06-16 150606](https://github.com/nevence/VacaYAY/assets/155846118/4e2033f3-c067-45de-867f-277428d62db7)
![Screenshot 2024-06-16 150828](https://github.com/nevence/VacaYAY/assets/155846118/0930635e-b3a5-4cde-83bd-0f2090e066c3)
![Screenshot 2024-06-16 152442](https://github.com/nevence/VacaYAY/assets/155846118/fbb5b40e-623f-4673-9e91-3f0888f77f55)

nevence committed:
feat: #6 create admin request management page

created HR (admin) and Employee request management pages, implemented filtering by leave type, status, search by name, and page size
implemented UI and logic for requests approval and rejection, and adding (editing) comments (HR and Employee) / changing leave type (HR)

nevence commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Admin request management page](https://trello.com/c/9zSr6HxD/6-admin-request-management-page)

NinaLazarevicIng commented on VacaYAY/BusinessLogicLayer/Extensions/VacationRequestExtension.cs:
```diff
@@ -56,6 +57,19 @@ public static void MapToVacationRequestUpdate(this VacationRequest vacationReque
+        public static void MapVacationRequestApprove(this VacationRequest vacationRequest)
+        {
+            vacationRequest.Status = Enums.VacationRequestStatus.Accepted;
+            vacationRequest.Employee.DaysOffNumber -= (vacationRequest.EndDate - vacationRequest.StartDate).Days;
```
Days off for employee are also number of working days, so you can use your function GetWorkingDays here.

nevence replied:
great observation, i accidentally let that slide

nevence committed:
edit: correction of updating days off number

nevence committed:
edit: included option to edit leave starting and ending dates

nevence committed:
edit: verification of leave dates validity

nevence committed:
feat: #7 add pagination

implemented pagination, search, filtering by position when retrieving employees. added functionalities to see details, edit and delete an employee

nevence commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Add pagination](https://trello.com/c/K6T1jF9i/7-add-pagination)

pdavitkovic commented on VacaYAY/PresentationLayer/ActionFilters/NotificationFilter.cs:
```diff
@@ -0,0 +1,25 @@
+        }
+
+        public void OnResultExecuting(ResultExecutingContext context)
+        {
```
Seems like an overkill. 

You can try this: https://www.c-sharpcorner.com/blogs/alert-message-from-controller-view-using-javascript-alert-messagebox

nevence committed:
edt: correction after PR, removed filter for adding notification messages

implemented passing messages manually in controller
